processors/record_linker/processor.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import uuid
+import logging
+from .. import base
+logger = logging.getLogger(__name__)
+
+
+# Module API
+
+def process(conf, conn):
+
+    # Trial m2m
+    TRIAL_TABLES = [
+        'trials_conditions',
+        'trials_interventions',
+        'trials_locations',
+        'trials_organisations',
+        'trials_persons',
+        'trials_publications',
+    ]
+
+    # Unlink wrong linked records
+    # [will be reprocessed on next processors run]
+    count = 0
+    for record in base.helpers.iter_rows(conn, 'database', 'records', orderby='id'):
```

Comment:
As far as I can see, there's no need for `orderby='id'` here. Also, I haven't tested it, but you could maybe add `conn['database']['records'].columns.trial_id != None` to remove the records without `trial_id`.